backend/mutations.py

```diff
@@ -218,23 +220,26 @@ def mutate(self, info, user_data=None):
         settings = Settings()
         metrics = UserMetrics()
         
-        user = User(
-            # Need to eventually add username (if that's not the 'name') and password
-            name = user_data.name,
-            bio = user_data.bio,    # not required by model, will assume frontend filters input
-            email = user_data.email,
-            password = user_data.password,
-            profile_pic = user_data.profile_pic,
-            metrics = metrics,
-            achievements = [], # will likely want to add a hard coded initial achievement
-            personal_portfolio = portfolio,
-            groups = [],
-            settings = settings
-        )
-        checkAchievements(user)
-        user.save()
+        try:
+            user = User.objects.get(pk=user_data.email)
+        except Exception as e: # email not taken
+            user = User(
+                name = user_data.name,
+                bio = user_data.bio,
+                email = user_data.email,
+                password = user_data.password,
+                profile_pic = user_data.profile_pic,
+                metrics = metrics,
+                achievements = [],
+                personal_portfolio = portfolio,
+                groups = [],
+                settings = settings
+            )
+            checkAchievements(user)
+            user.save()
+            return CreateUserMutation(user=user, success=True)
```

Comment:
Actually, I feel like it might not be good practice to have our "normal" execution depend on an Exception being thrown. Could it be possible to use the `contains` method instead to check for email?

Reply:
You're absolutely right! I will look into it, but this shouldn't cause any problems at the moment.

Reply:
(For documentation) an issue was made for this, but the PR was pushed since this edit functionally does what we want. This requires separate research into what could be returned.